protocol/0014-ORDT-order_types.md

```diff
@@ -47,8 +47,61 @@ Notes on scope of current version of this spec:
 ### Execution flags
 
 1. **Post-Only (True/False):** Only valid for Limit orders. Cannot be True at the same time as Reduce-Only. If set to true, once order reaches the orderbook, this order acts identically to a limit order set at the same price. However, prior to being placed a check is run to ensure that the order will not (neither totally nor in any part) immediately cross with anything already on the book. If the order would immediately trade, it is instead immediately `Stopped` with a reason informing the trader that the order was stopped to avoid a trade occurring. As a result, placing a Post-Only order will never incur taker fees, and will not incur fees in general if executed in continuous trading. It is possible for some liquidity and infrastructure fees to be paid if the resultant limit order trades at the uncrossing of an auction, as specified in [0029-FEES](https://github.com/vegaprotocol/specs/blob/master/protocol/0029-FEES-fees.md#normal-auctions-including-market-protection-and-opening-auctions).
+
 1. **Reduce-Only (True/False):** Only valid for Non-Persistent orders. Cannot be True at the same time as Post-Only. If set, order will only be executed if the outcome of the trade moves the trader's position closer to 0. In addition, a Reduce-Only order will not move a position to the opposite side to the trader's current position (e.g. if short, a Reduce-Only order cannot make the trader long as a result). If submitted as IOC, where the full volume would switch sides, only the amount required to move the position to 0 will be executed.
 
+
+### Stop orders
+
+In addition to normal immediately executing order, Vega should accept the submission of stop orders.
+These differ from normal orders in that they sit off the order book until triggered, when they are entered as normal.
+These are generally used to exit positions under pre-defined conditions, either as a "stop loss" order that controls the maximum losses a position may take, a "take profit" order that closes a position once a defined level of profit has been made, or both.
+To prevent traders from "hiding" order book depth behind conditional orders, stop orders can only be used to close some or all of a trader's position, and therefore must be "reduce only" orders.
+
+A stop order submission can be made (stop loss or take profit are probably both just called a stop order internally).
+
+* Stop order submissions must include either a trigger price OR trailing stop distance as a % move from the reference price in addition to a normal order submission.
+
+* Stop order submissions must include a trigger direction.
+Direction may be _rises above_ or _falls below_.
+_Rises above_ stops trigger if the last traded price is higher than the trigger level, and _falls below_ stops trigger if the last traded price is lower than the trigger level.
```

Comment:
Would it take a lot away from the feature if we made this conditional on mark price instead to save some computation? Can of course start with last traded price and modify it if does become an issue.

Reply:
I think it isn't great to do that, would be quite surprising for users I think. Let's start with last traded price, stops shouldn't be a huge amount of computation to evaluate as you can maintain them in a sorted data structure.